core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -5958,6 +5999,13 @@ private void checkLiteral2(String expression, String expected) {
         .ok(expected);
   }
 
+  @Test void testDuckDBDecimalPrecision() {
+    final String sql = "SELECT CAST(1.23 AS DECIMAL(38, 37))";
+    final String expected = "SELECT 1.2300000000000000000000000000000000000";
```

Comment:
I think this test would be more robust if the last digit was not 0
you are relying on the formatting to print zeros which are not significant, and this may not always be true

Reply:
From definition：
LAST：Nulls last for DESC and ASC.
HIGH：Nulls first for DESC, nulls last for ASC.

system env test：
before query
insert into t3 select 1,'aa';
insert into t3 select 1,'bb';
insert into t3 select 1,NULL;
insert into t3 select 1,NULL;

then query order by desc/asc
D select * from t3 order by data asc;
┌───────┬─────────┐
│  id   │  data   │
│ int32 │ varchar │
├───────┼─────────┤
│     1 │ aa      │
│     1 │ bb      │
│     1 │         │
└───────┴─────────┘
D select * from t3 order by data desc;
┌───────┬─────────┐
│  id   │  data   │
│ int32 │ varchar │
├───────┼─────────┤
│     1 │ bb      │
│     1 │ aa      │
│     1 │         │
└───────┴─────────┘

we can find that NULL is always at last, so here should be LAST.

Reply:
Thanks for the reminder, had add a case with non-zero at last. @mihaibudiu